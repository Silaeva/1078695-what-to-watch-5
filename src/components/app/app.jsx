import React from "react";
import {Switch, Route, BrowserRouter} from "react-router-dom";
import MainPage from "../main-page/main-page";
import SignIn from "../sign-in/sign-in";
import MyList from "../my-list/my-list";
import FilmPage from "../film-page/film-page";
import AddReview from "../add-review/add-review";
import VideoPlayer from "../player/player";

import withFullscreenPlayer from "../../hocs/with-fullscreen-player/with-fullscreen-player";

const Player = withFullscreenPlayer(VideoPlayer);

const App = () => {
  return (
    <BrowserRouter>
      <Switch>
        <Route exact
          path="/"
          render={({history}) => (
            <MainPage
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
            />
          )} />
        <Route exact path="/login">
          <SignIn />
        </Route>
        <Route exact
          path="/mylist"
          render={({history}) => (
            <MyList
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
            />
          )}
        />
        <Route exact
          path="/films/:id"
          render={({history, match}) => (
            <FilmPage
              currentFilmId={match.params.id}
              onFilmCardClick={(id) => history.push(`/films/${id}`)}
            />
          )}
        >
        </Route>
        <Route exact
          path="/films/:id/review"
          render={({match}) => (
            <AddReview
              currentFilmId={match.params.id}
            />
          )}
        />
        <Route exact
          path="/player/:id"
          render={({match}) => (
            <Player
              currentFilmId={match.params.id}
            />
          )}
        >
        </Route>
      </Switch>
    </BrowserRouter>
  );
};

export default App;
