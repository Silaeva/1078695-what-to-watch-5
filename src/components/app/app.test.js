import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app";

jest.mock(`../loading/loading`, () => `Loading`);
jest.mock(`../error-screen/error-screen`, () => `ErrorScreen`);
jest.mock(`../main-page/main-page`, () => `MainPage`);
jest.mock(`../sign-in/sign-in`, () => `SignIn`);
jest.mock(`../my-list/my-list`, () => `MyList`);
jest.mock(`../film-page/film-page`, () => `FilmPage`);
jest.mock(`../add-review/add-review`, () => `AddReview`);
jest.mock(`../player/player`, () => `VideoPlayer`);
jest.mock(`../private-route/private-route`, () => `PrivateRoute`);

describe(`Should App render correctly`, () => {
  it(`With loading success`, () => {
    const tree = renderer
      .create(
          <App
            isLoading={false}
            isLoadingError={false}
            authInProgress={false}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With authorization in progress`, () => {
    const tree = renderer
      .create(
          <App
            isLoading={false}
            isLoadingError={false}
            authInProgress={true}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`When page loading`, () => {
    const tree = renderer
      .create(
          <App
            isLoading={true}
            isLoadingError={false}
            authInProgress={false}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With page loading error`, () => {
    const tree = renderer
      .create(
          <App
            isLoading={false}
            isLoadingError={true}
            authInProgress={false}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
