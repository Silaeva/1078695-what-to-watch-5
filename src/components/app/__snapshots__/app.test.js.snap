// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should App render correctly When page loading 1`] = `
<MainPage
  onFilmCardClick={[Function]}
/>
`;

exports[`Should App render correctly With authorization in progress 1`] = `
<MainPage
  onFilmCardClick={[Function]}
/>
`;

exports[`Should App render correctly With loading success 1`] = `
<MainPage
  onFilmCardClick={[Function]}
/>
`;

exports[`Should App render correctly With page loading error 1`] = `<ErrorScreen />`;
